### PR TITLE
[Fix] Synchronize frame initializers for local_var and local_scope & Remove warning

### PR DESCRIPTION
**Pull request by mufeili**
## Description
Currently `local_var` and `local_scope` do not use the frame initializers of the original `_node_frame` and `_edge_frame`. This causes warnings about initializers even when we have initializers set, which can be confusing for example when using `nn modules`.

Also as discussed before, warnings about frame initializers are removed.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

## Changes
- Synchronize frame initializers after the copy of frames
- Test cases are added

### CODE REVIEW

File: python/dgl/graph.py [hunk 1]
```diff
@@ -3451,6 +3455,24 @@ def local_scope(self):
         old_eframe = self._edge_frame
         self._node_frame = FrameRef(Frame(self._node_frame._frame))
         self._edge_frame = FrameRef(Frame(self._edge_frame._frame))
+        sync_frame_initializer(self._node_frame._frame, old_nframe._frame)
+        sync_frame_initializer(self._edge_frame._frame, old_eframe._frame)
         yield
         self._node_frame = old_nframe
         self._edge_frame = old_eframe
+
+def sync_frame_initializer(new_frame, reference_frame):
```
Comment (yzh119) [inline python/dgl/graph.py]: I recommend moving this function to frame.py

Comment (mufeili) [reply python/dgl/graph.py]: Done.

File: python/dgl/graph.py [hunk 2]
```diff
@@ -3401,9 +3401,13 @@ def local_var(self):
         DGLGraph
             The graph object that can be used as a local variable.
         """
+        local_node_frame = FrameRef(Frame(self._node_frame._frame))
```
Comment (yzh119) [inline python/dgl/graph.py]: Better explain the behavior of initializer in the docstring.

File: python/dgl/graph.py [hunk 3]
```diff
@@ -3451,6 +3455,8 @@ def local_scope(self):
         old_eframe = self._edge_frame
         self._node_frame = FrameRef(Frame(self._node_frame._frame))
         self._edge_frame = FrameRef(Frame(self._edge_frame._frame))
+        sync_frame_initializer(self._node_frame._frame, old_nframe._frame)
```
Comment (yzh119) [inline python/dgl/graph.py]: The same.

**Comment by jermainewang** [comment]
Hey @mufeili, if we just get rid of the warning, could it be easier? I suppose we only need to change `warn_and_set_initializer`?

File: python/dgl/graph.py (hunk 2 above)
Comment (mufeili) [reply python/dgl/graph.py]: Better?

File: python/dgl/graph.py (hunk 3 above)
Comment (mufeili) [reply python/dgl/graph.py]: Better?

**Comment by mufeili** [comment]
@jermainewang I've removed the warnings. However, I think this synchronization is still necessary as long as we do not completely deprecate initializers.

File: python/dgl/graph.py (hunk 2 above)
Comment (yzh119) [reply python/dgl/graph.py]: I mean users should know the `local_var` use the same node/edge initializer as the original `DGLGraph`, maybe you could mention it in the Notes section.

Comment (mufeili) [reply python/dgl/graph.py]: Done.

**Comment by mufeili** [comment]
@yzh119 Any further concerns?

**Comment by yzh119** [comment]
Please go ahead and merge it.

**Pull request closed without merging**